cmake/Modules/FindTileDB_EP.cmake

```diff
@@ -51,8 +51,15 @@ if (NOT TILEDB_FOUND)
           SET(DOWNLOAD_URL "https://github.com/TileDB-Inc/TileDB/releases/download/2.16.3/tiledb-windows-x86_64-2.16.3-194b5ae.zip")
           SET(DOWNLOAD_SHA1 "8e986b6143ef9509201c7f2b3c0a7a53ff41095a")
         elseif(APPLE) # macOS
-          SET(DOWNLOAD_URL "https://github.com/TileDB-Inc/TileDB/releases/download/2.16.3/tiledb-macos-x86_64-2.16.3-194b5ae.tar.gz")
-          SET(DOWNLOAD_SHA1 "b4d35306771d1c30b49bf2a98651e8e24c91037e")
+            if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
```

Comment:
This won't work correctly if we are "cross-compiling" from x86_64, we need to also check `CMAKE_OSX_ARCHITECTURES`. See: https://github.com/TileDB-Inc/TileDB-Vector-Search/commit/b3c70a957a1d20ab3be79923c2447c6b90c93948

Reply:
Thanks!